### PR TITLE
fix for issue 207

### LINKED ISSUE #207: When logging in after logging out there is thrown an exception when the loading spinner is popped
**Describe the bug**
If you log out and then log in again there is thrown an issue because the loading spinne widget is pop'ed with a outdated context. 

**To Reproduce**
1. Go to choose citizen screen
2. Click on log out button
3. Log in again
4. See exception

**Expected behavior**
It doesn't throw an exception

**Actual behavior**
it throws an exception

### PR DESCRIPTION
**Pull request by antonhpalmer**
fixes #207

### CODE REVIEW

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw607f20 you have been assigned this. Here's your checklist. 
Please use the following checklist. Feel free to check it off as you go along.

**Code Design**
- [x] Is the code in the right place? (Both in terms of folder structure and class structure)
- [ ] Could this code, to the best of your knowledge, have reused existing code?
- [ ] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [ ] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [x] Are names meaningful and self-documenting?
- [x] Is it understandable, by reading the code, what it does?
- [x] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [ ] Is it understandable, by reading the tests, what they do?
- [ ] Do the tests cover sensible cases? Both happy paths and exception paths?
- [ ] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [x] Are error messages in the code understandable?
- [x] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [x] Are the features that the code claims to implement, actually implemented?
- [ ] Does it look like the code has bugs?
- [x] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [ ] Are UI messages, that the user might meet, actually helpful?
- [ ] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [x] Is there well written code in this? Remember to tell the developer!

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw614f20 you have been assigned this. Here's your checklist. 
Please use the following checklist. Feel free to check it off as you go along.

**Code Design**
- [ ] Is the code in the right place? (Both in terms of folder structure and class structure)
- [ ] Could this code, to the best of your knowledge, have reused existing code?
- [ ] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [ ] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [ ] Are names meaningful and self-documenting?
- [ ] Is it understandable, by reading the code, what it does?
- [ ] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [ ] Is it understandable, by reading the tests, what they do?
- [ ] Do the tests cover sensible cases? Both happy paths and exception paths?
- [ ] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [ ] Are error messages in the code understandable?
- [ ] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [ ] Are the features that the code claims to implement, actually implemented?
- [ ] Does it look like the code has bugs?
- [ ] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [ ] Are UI messages, that the user might meet, actually helpful?
- [ ] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [ ] Is there well written code in this? Remember to tell the developer!

**Comment by joandrsn** [comment]
@antonhpalmer 
Is this not a violation of the BLoC pattern?
As I understand the [BLoC pattern](http://flutterdevs.com/blog/bloc-pattern-in-flutter-part-1/), you should seperate UI logic from business logic. And with this, you have bound changes to the UI to the `auth_bloc`.
I might be wrong, but if I am, could you please give some reasoning why?

Another thing: Would it be a possibility to make a test of this in order to make sure that no exception is thrown? (A test which would have caught the issue and with this change is now successful)

**Comment by antonhpalmer** [comment]
@joandrsn 
You are totally right, the context should not be passed to a BLoC. I looked further into the underlying issue and found the actual problem. The observable `authBloc.loggedIn` that we listen to in `login_screen.dart` was for some reason changed to `true` twice everytime a login occured after the first login. 

I therefore just added a bool `_popcalled` to make sure that pop is only called after the first change on the observable.

I am not sure exactly how to test that the observable is changed twice but pop is only called once. 
Will look into it.

**Comment by antonhpalmer** [comment]
We think a test for this is unnecessary since it is a very specific exception. 
Please re-review @joandrsn @aau-giraf/sw607f20 @aau-giraf/sw614f20

**Review by tdybda17 — DISMISSED** [review]

File: lib/blocs/auth_bloc.dart [hunk 1]
```diff
@@ -3,6 +3,7 @@ import 'package:rxdart/rxdart.dart';
 import 'package:weekplanner/blocs/bloc_base.dart';
 import 'package:weekplanner/models/enums/weekplan_mode.dart';
 
+
```
Comment (joandrsn) [inline lib/blocs/auth_bloc.dart]: Please remove this whitespace.

**Review by joandrsn — APPROVED** [review]

**Review by tdybda17 — APPROVED** [review]

**Pull request merged**